chaos_kitten/cli.py

```
@@ -185,12 +191,35 @@ def scan(
     # Run the orchestrator
     from chaos_kitten.brain.orchestrator import Orchestrator
     orchestrator = Orchestrator(app_config)
     try:
         import asyncio
-        asyncio.run(orchestrator.run())
+        results = asyncio.run(orchestrator.run())
+
+        # Check for orchestrator runtime errors
+        if isinstance(results, dict) and results.get("status") == "failed":
+            console.print(f"[bold red]❌ Scan failed:[/bold red] {results.get('error')}")
+            raise typer.Exit(code=1)
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)